lib/mayaUsd/ufe/UsdUndoAddNewPrimCommand.h

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright 2020 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef USD_ADD_NEW_PRIM_COMMAND
+#define USD_ADD_NEW_PRIM_COMMAND
+
+#include <ufe/undoableCommand.h>
+#include <ufe/path.h>
+
+#include <mayaUsd/base/api.h>
+#include <mayaUsd/ufe/UsdSceneItem.h>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+MAYAUSD_NS_DEF {
+namespace ufe {
+
+//! \brief Undoable command for add new prim
+class MAYAUSD_CORE_PUBLIC UsdUndoAddNewPrimCommand : public Ufe::UndoableCommand
+{
+public:
+    typedef std::shared_ptr<UsdUndoAddNewPrimCommand> Ptr;
+
+    UsdUndoAddNewPrimCommand(const UsdSceneItem::Ptr& usdSceneItem,
+                              const std::string& name,
+                              const std::string& type);
+
+	void undo() override;
+	void redo() override;
+
+    const Ufe::Path& newUfePath() const;
+    
+	static UsdUndoAddNewPrimCommand::Ptr create(const UsdSceneItem::Ptr& usdSceneItem,

```

Comment:
Actually I'm voting to leave as-is.  All the other command classes have the full namespace listed here (so this one would look a bit odd if it was the only one that didn't), and I actually like it because there are multiple "Ptr" typedefs being used (e.g. this class has a Ptr and there's also the scene item Ptr's being used in the same functions).

Reply:
O.K.